### PR TITLE
fix warnings in BlurHashDecoder.kt

### PR DESCRIPTION
**Pull request by connyduck**

### CODE REVIEW

File: Kotlin/lib/src/main/java/com/wolt/blurhashkt/BlurHashDecoder.kt [hunk 1]
```diff
@@ -21,8 +21,6 @@ object BlurHashDecoder {
         '|', '}', '~'
     )
 
-    private val MAX_SIZE = 20
```
Comment (connyduck) [inline Kotlin/lib/src/main/java/com/wolt/blurhashkt/BlurHashDecoder.kt]: unused

File: Kotlin/lib/src/main/java/com/wolt/blurhashkt/BlurHashDecoder.kt [hunk 2]
```diff
@@ -2,7 +2,7 @@ package com.wolt.blurhashkt
 
 import android.graphics.Bitmap
 import android.graphics.Color
-import java.lang.Math.*
+import kotlin.math.*
```
Comment (connyduck) [inline Kotlin/lib/src/main/java/com/wolt/blurhashkt/BlurHashDecoder.kt]: Its Kotlin, so `kotlin.math` should be used

File: Kotlin/lib/src/main/java/com/wolt/blurhashkt/BlurHashDecoder.kt [hunk 3]
```diff
@@ -32,14 +30,14 @@ object BlurHashDecoder {
         val numY = (sizeFlag / 9) + 1
         val numX = (sizeFlag % 9) + 1
 
-        val quantisedMaximumValue = blurHash[1].toString().decode83()
-        val maximumValue = (quantisedMaximumValue + 1) / 166f
-
         if (blurHash.length != 4 + 2 * numX * numY) {
             return null
         }
```
Comment (connyduck) [inline Kotlin/lib/src/main/java/com/wolt/blurhashkt/BlurHashDecoder.kt]: fail early - move the computations that are not needed to detect this case down

File: Kotlin/lib/src/main/java/com/wolt/blurhashkt/BlurHashDecoder.kt [hunk 4]
```diff
@@ -94,30 +91,30 @@ object BlurHashDecoder {
...
-                signPow((quantG - 9) / 9.0, 2.0) * maximumValue,
-                signPow((quantB - 9) / 9.0, 2.0) * maximumValue
+                signPow((quantR - 9) / 9.0f, 2.0f) * maximumValue,
+                signPow((quantG - 9) / 9.0f, 2.0f) * maximumValue,
+                signPow((quantB - 9) / 9.0f, 2.0f) * maximumValue
         )
     }
 
-    private fun signPow(value: Double, exp: Double) = copySign(pow(abs(value), exp), value).toFloat()
+    private fun signPow(value: Float, exp: Float) = value.pow(exp).withSign(value)
```
Comment (connyduck) [inline Kotlin/lib/src/main/java/com/wolt/blurhashkt/BlurHashDecoder.kt]: `abs()` is not needed. Yes, the sign may be different after the `pow`, but `withSign` (Kotlin for `copySign`) changes it again anyway.

File: Kotlin/lib/src/main/java/com/wolt/blurhashkt/BlurHashDecoder.kt [hunk 5]
```diff
@@ -75,10 +73,9 @@ object BlurHashDecoder {
         return bitmap
     }
 
-    @Suppress("LoopToCallChain")
```
Comment (connyduck) [inline Kotlin/lib/src/main/java/com/wolt/blurhashkt/BlurHashDecoder.kt]: What is it suppressing? I can remove this line without getting more warnings.

File: Kotlin/lib/src/main/java/com/wolt/blurhashkt/BlurHashDecoder.kt (hunk 1 above)
Comment (hangduykhiem) [reply Kotlin/lib/src/main/java/com/wolt/blurhashkt/BlurHashDecoder.kt]: The value is recommended in the `Readme.md`. It is unused for now, but it should be public and used [here](https://github.com/woltapp/blurhash/blob/6a0abe6e2f57d3960b23d8f0fcece2bc27d09ac9/Kotlin/demo/src/main/java/com/wolt/blurhashapp/MainActivity.kt#L28). If you could update that line, the PR is ready to go. Good stuff!

Comment (connyduck) [reply Kotlin/lib/src/main/java/com/wolt/blurhashkt/BlurHashDecoder.kt]: Done!

**Pull request merged**